src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -54,8 +58,18 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
                 // it's a received message, do not send it back
                 $this->logger->info('Received message "{class}"', $context);
             } else {
+                $messageDispatched = false;
```

Comment:
Instead of this boolean, can you extract the `sendersLocator->getSenders()` call into a `$senders` variable and have this `dispatch` within a `if (\count($senders) > 0) {`? It would be much more readable.